### ISSUE TITLE
Should inherit from AbstractPoolCache

### ISSUE THREAD

**Issue by alarofrunetotem** [php-cache/issues#113]
If you have an expensive cache and later decide to mix it with a faster cache (i.e.filesystem and memcache) you cant just plug in this class as a drop in replacement because it does not inherits from AbstractPoolCache. It also misses the PSR-16 compatibility (beause it's implemented in AbstractPoolCache.

**Comment by prisis** [comment]
Hey @alarofrunetotem can you tell my the name of the class, i dont see which class it is?

**Comment by ggargani** [comment]
(Writing from my work account now)
Sorry, I didnt notice issues are at php-cache level (and not for single driver).
I was referring to the chain-adapter.

**Comment by hytest** [comment]
I have some problem on NamespacedCachePool,  it doesn't inherit from AbstractPoolCache so I can't call invalidateTags.

**Comment by prisis** [comment]
Yeah the NamespacedCachePool is missing some functions, i will work this weekend on phpcache, but im happy about a pr

**Comment by cryptiklemur** [comment]
Thanks for clarifying the affected classes. [`cache/chain-adapter` 2.0.3](https://github.com/php-cache/chain-adapter/blob/2.0.3/CachePoolChain.php) implements PSR-6, PSR-16, and tag invalidation directly. [`NamespacedCachePool::create()`](https://github.com/php-cache/namespaced-cache/blob/2.0.2/NamespacedCachePool.php#L43-L52) preserves tag support when the wrapped pool supports it. These capabilities no longer depend on inheriting from `AbstractCachePool`. I'm closing this as resolved. If one of these replacement paths is still missing an operation, please reopen the issue with a reproducer.
